helm_deploy/demo-app/templates/deployment.yaml

```diff
@@ -0,0 +1,21 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: django
+  labels:
+    app: django
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: django-container
+  template:
+    metadata:
+      labels:
+        app: django-container
+    spec:
+      containers:
+        - name: django
```

Comment:
Currently this will be running the django app using `manage.py runserver`, which is [not suitable for production](https://docs.djangoproject.com/en/2.0/ref/django-admin/#runserver).

Best practice is to put a webserver in front of Django and proxy web application HTTP requests back to the python WSGI app, and serve static assets directly from the filesystem - most commonly [using nginx](https://uwsgi-docs.readthedocs.io/en/latest/tutorials/Django_and_nginx.html).

This means that the Pod here will now contain two containers, nginx and django. As the lifecycles of containers within a Pod are bound together, they will always be deployed on the same host, will share `localhost`, and can share a filesystem, which allows the use of UNIX sockets for web server->WSGI communication rather than HTTP.

This will require some configuration for nginx, to define the application upstream - this can be handled as a [ConfigMap object mounted as a Volume](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#add-configmap-data-to-a-volume) into the nginx container.

Reply:
I'm not sure if I'm only complicating this, but I feel like I need some support from the team tomorrow when I'm in.

Reply:
Sure, this is relatively complicated - let's have a chat tomorrow.

Reply:
Fair enough - the question of how production-like apps should be in non-production environments is a matter of opinion really. I'd argue that they should generally be as production-like as possible, and in this context it'd both test the platform more fully and our user's understanding of what's required of them, and what's possible. This isn't a cut and dried thing though - thoughts @kalbir ?